soccer_strategy/src/soccer_strategy/robot_controlled_2d.py

```diff
@@ -77,3 +79,25 @@ def observe_ball(self, ball: Optional[Ball]) -> bool:
             ball.last_observed_time_stamp = rospy.Time.now()
 
         # TODO can add noise here
+
+    def observe_obstacles(self, robots: List[Robot]):
+        self.observed_obstacles.clear()
```

Comment:
Instead of clearing, checks if the observed obstacles detection time compared to current time is less than 10 seconds, and clear if it is greater than 10 seconds